mlflow/utils/__init__.py

```diff
@@ -53,3 +58,43 @@ def reraise(tp, value, tb=None):
     finally:
         value = None
         tb = None
+
+
+def chunk_dict(d, chunk_size):
+    # Copied from: https://stackoverflow.com/a/22878842
+
+    it = iter(d)
+    for _ in range(0, len(d), chunk_size):
+        yield {k: d[k] for k in islice(it, chunk_size)}
+
+
+def truncate_dict(d, max_key_length=None, max_value_length=None):
```

Comment:
Could you make sure we have an internal follow-up JIRA for deduplicating some of this code between pyspark & sklearn autologging? e.g:

* Pulling the ``TrainingSession`` abstraction (shared between sklearn and pyspark.ml) into `mlflow/utils/autologging_utils/training_session.py` and updating sklearn & pyspark.ml autologging to use the shared abstraction (deleting the separate `SparkTrainingSession` and `SklearnTrainingSession` classes)
* Similarly, deduplicating the `_truncate_dict` method defined in `mlflow/sklearn/utils.py` with the new shared method defined in this file (delete `_truncate_dict` from mlflow/sklearn/utils.py and call the `mlflow.utils.truncate_dict` API` instead in sklearn autologging)

Are there other things we should make sure to deduplicate?

Reply:
I think that's all and here's a PR to address this: https://github.com/mlflow/mlflow/pulls